src/main/java/com/google/roomies/data/Comment.java

```diff
@@ -0,0 +1,67 @@
+package com.google.roomies;
+ 
+import static com.google.roomies.CommentRequestParameterNames.COMMENT;
+import static com.google.roomies.CommentRequestParameterNames.TIMESTAMP;
+
+import com.google.api.core.ApiFuture;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.Timestamp;
+import com.google.common.collect.ImmutableMap;
+import com.google.roomies.database.DatabaseFactory;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import javax.servlet.http.HttpServletRequest;
+
+/** A comment made by a user under a listing. */
+@AutoValue
+public abstract class Comment {
+  abstract Optional<String> documentId();
```

Comment:
It's the comment ID in the database. I think it would be good to keep it for future functionality (editing comment, deleting comment, etc) as the ID is the only way to locate the comment in the database.

Renamed to commentId to make it clearer.